server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestClusterHealthAction.java

```diff
@@ -48,7 +51,17 @@ public boolean allowSystemIndexAccessByDefault() {
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
         final ClusterHealthRequest clusterHealthRequest = fromRequest(request);
-        return channel -> client.admin().cluster().health(clusterHealthRequest, new RestStatusToXContentListener<>(channel));
+        return channel -> {
+            ActionListener<ClusterHealthResponse> listener = new RestStatusToXContentListener<>(channel);
```

Comment:
Rather than checking here, `ActionRequest` has a `validate` method that checks if the request is valid, the checking should be done there. eg for `RestClusterHealthAction`, its in the `ClusterHealthRequest.validate()` method.